src/transformers/trainer.py

```diff
@@ -240,6 +244,22 @@ def __init__(
         if model is None and model_init is not None:
             model = self.call_model_init()
 
+        if args.deepspeed:
+            model_parameters = filter(lambda p: p.requires_grad, model.parameters())
+            model, optimizer, training_dataloader, lr_scheduler = deepspeed.initialize(
+                args=args,
+                model=model,
+                model_parameters=model_parameters,
+                # optimizer=optimizer,
```

Comment:
are those commented-out statements not needed anymore?

Reply:
These are options I haven't explored yet, so they are there to see that this can be done.
```
                # optimizer=optimizer,
                # lr_scheduler=lr_scheduler,
                # training_data=trainset,
```

The first 2 are there in case DS doesn't have a particular scheduler/optimizer in its toolkit, but I haven't tried these yet.

Wrt the third one it seems that our trainer already handles the batching, so I'm not sure if we need to delegate this feature to DS or not.

Unlike fairscale's sharded features, DS has dozens of features, and exploring each is a process on its own. My main intention was to get the bulk off the road and then multiple devs can explore various sub-features.